packages/ethereum-contracts/truffle-config.js

```diff
@@ -27,31 +27,66 @@ try {
     );
 }
 
+const ALIASES = {
+    "eth-mainnet": ["mainnet"],
+    "eth-ropsten": ["ropsten"],
+    "eth-rinkeby": ["rinkeby"],
+    "eth-kovan": ["kovan"],
+    "eth-goerli": ["goerli"],
+
+    "xdai-mainnet": ["xdai"],
+
+    "polygon-mainnet": ["matic"],
+    "polygon-mumbai": ["mumbai"],
+
+    "optimism-mainnet": ["opmainnet"],
+    "optimism-kovan": ["opkovan"],
+
+    "arbitrum-one": ["arbone"],
+    "arbitrum-rinkeby": ["arbrinkeby"],
+
+    "avalanche-C": ["avalanche"],
+    "avalanche-fuji": ["avafuji"],
+
+    "bsc-mainnet": ["bsc"],
+
+    "celo-mainnet": ["celo"],
+};
+
 const DEFAULT_NETWORK_TIMEOUT = 60000;
 
-//
-// This is a hack to resolve that HDWallet doesn't work with openethereum (xdai & kovan)
-//
-// Related issues:
-// - https://github.com/trufflesuite/truffle/issues/3182
-// - https://github.com/openethereum/parity-ethereum/issues/11824
-// - https://github.com/MetaMask/web3-provider-engine/issues/311
-function createProviderForOpenEthereum(url) {
-    let provider;
-    const Web3WsProvider = require("web3-providers-ws");
-    if (url.startsWith("ws:") || url.startsWith("wss:")) {
-        provider = new Web3WsProvider(url);
-        // apply the skipCache hack
-        const origSend = provider.__proto__.send;
-        provider.__proto__.send = function (payload, callback) {
-            delete payload.skipCache;
-            origSend.call(provider, payload, callback);
-        };
-    } else {
-        // let hdwallet provider handle the url directly
-        provider = url;
-    }
-    return provider;
+/**
+ * Get the environment value for the network, supporting network aliases
+ *
+ * E.g. for MNEMONIC key, these environment keys are tried (high to low priority):
+ * ETH_MAINNET_MNEMONIC
+ * MAINNET_MNEMONIC
+ * DEFAULT_MNEMONIC,
+ */
+function getEnvValue(networkName, key) {
+    const keysToTry = [networkName, ...ALIASES[networkName]]
+        .map((i) => i.replace(/-/g, "_").toUpperCase() + "_" + key)
+        .concat("DEFAULT_" + key);
+    const values = keysToTry.map((i) => process.env[i]).filter((i) => !!i);
+    return values[0];
+}
+
+/**
+ * Create default network configurations
+ */
+function createNetworkDefaultConfiguration(networkName, chainId) {
+    return {
+        provider: () =>
+            new HDWalletProvider({
+                mnemonic: getEnvValue(networkName, "MNEMONIC"),
```

Comment:
Note that this change breaks configs using private keys with truffle.  
The legacy initialization method accepted both passphrase or private key as first argument. With the new way, the field `privateKeys` needs to be set for PKs.

Reply:
are you using the privateKey initialization method?

can yhou show an example?

Reply:
Currently I just assign the PK to the mnemonic var, e.g. `XDAI_MNEMONIC=123...`.
That works with the [legacy way of initializing HDWalletProvider](https://github.com/trufflesuite/truffle/tree/develop/packages/hdwallet-provider#using-the-legacy-interface-deprecated), bcs the first argument could be mnemonic, pk, or pk array.